keyboards/keychron/q3/matrix.c

```diff
@@ -0,0 +1,192 @@
+/* Copyright 2021 @ Keychron (https://www.keychron.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "matrix.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "debounce.h"
+#include "quantum.h"
+#include "util.h"
```

Comment:
I think that most of these can be removed safely.

Reply:
Thanks, I tested that it is ok to only keep matrix.h and quantum.h.